### PR TITLE
LCORE-2400: Unit tests for schema dumper

### PR DESCRIPTION
**Pull request by tisnik**
## Description

LCORE-2400: Unit tests for schema dumper

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up service version
- [ ] Bump-up dependent library
- [ ] Bump-up library or tool used for development (does not change the final image)
- [ ] CI configuration change
- [ ] Konflux configuration change
- [x] Unit tests improvement
- [ ] Integration tests improvement
- [ ] End to end tests improvement
- [ ] Benchmarks improvement

## Tools used to create PR

- Assisted-by: N/A
- Generated by: N/A

## Related Tickets & Documents

- Related Issue #LCORE-2400

## Release Notes

* **Tests**
  * Added five new unit tests for schema utilities covering empty inputs, simple schemas, nested properties, and edge case handling.

### CODE REVIEW

File: tests/unit/utils/test_schema_dumper.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    original: dict[str, Any] = {
+        "type": "object",
+        "properties": {
+            "name": {"type": "string"},
+            "age": {"type": "integer"},
+        },
+    }
+
+    # we need to distinguish between original and a copy
+    expected = original.copy()
```
[the remark this answers is not in the document]
Comment (tisnik) [reply tests/unit/utils/test_schema_dumper.py:54]: @coderabbitai create issue
